### PR TITLE
add batch infer for cpp rec

### PR DESCRIPTION
**Pull request by MissPenguin**

### CODE REVIEW

File: deploy/cpp_infer/include/preprocess_op.h [hunk 1]
```diff
@@ -44,6 +44,11 @@ class Permute {
   virtual void Run(const cv::Mat *im, float *data);
 };
 
+class Permute_batch {
```
Comment (littletomatodonkey) [inline deploy/cpp_infer/include/preprocess_op.h:47]: PermuteBatch

File: deploy/cpp_infer/src/ocr_rec.cpp [hunk 2]
```diff
@@ -15,83 +15,109 @@
...
+        auto preprocess_start = std::chrono::steady_clock::now();
+        int end_img_no = min(img_num, beg_img_no + this->rec_batch_num_);
+        float max_wh_ratio = 0;
+        for (int ino = beg_img_no; ino < end_img_no; ino ++) {
+            int h = img_list[indices[ino]].rows;
+            int w = img_list[indices[ino]].cols;
+            float wh_ratio = w * 1.0 / h;
+            max_wh_ratio = max(max_wh_ratio, wh_ratio);
+        }
+//         cout << "max_wh_ratio: " << max_wh_ratio << endl;
```
Comment (littletomatodonkey) [inline deploy/cpp_infer/src/ocr_rec.cpp:41]: 注释删掉吧

File: deploy/cpp_infer/src/preprocess_op.cpp [hunk 3]
```diff
@@ -40,6 +40,17 @@ void Permute::Run(const cv::Mat *im, float *data) {
   }
 }
 
+void Permute_batch::Run(const std::vector<cv::Mat> imgs, float *data) {
```
Comment (littletomatodonkey) [inline deploy/cpp_infer/src/preprocess_op.cpp:43]: PermuteBatch同上

File: deploy/cpp_infer/include/preprocess_op.h (hunk 1 above)
Comment (MissPenguin) [reply deploy/cpp_infer/include/preprocess_op.h]: done, thanks

File: deploy/cpp_infer/src/ocr_rec.cpp (hunk 2 above)
Comment (MissPenguin) [reply deploy/cpp_infer/src/ocr_rec.cpp]: done, thanks

File: deploy/cpp_infer/src/preprocess_op.cpp (hunk 3 above)
Comment (MissPenguin) [reply deploy/cpp_infer/src/preprocess_op.cpp]: done, thanks

**Review by tink2123 — APPROVED** [review]
LGTM

**Pull request merged**